### PR TITLE
Add missing const attribute to Public Key API

### PR DESCRIPTION
**Pull request by mstarzyk-mobica**
Partially implements #4033, where possible - add const attribute to public key api functions' parameters (in scope of 3.0 release).

## Status
**READY**

### CODE REVIEW

File: include/mbedtls/pk.h [hunk 1]
```diff
@@ -760,7 +760,7 @@ int mbedtls_pk_parse_public_keyfile( mbedtls_pk_context *ctx, const char *path )
  * \return          length of data written if successful, or a specific
  *                  error code
  */
-int mbedtls_pk_write_key_der( mbedtls_pk_context *ctx, unsigned char *buf, size_t size );
+int mbedtls_pk_write_key_der(const mbedtls_pk_context *ctx, unsigned char *buf, size_t size );
```
Comment (mpg) [inline include/mbedtls/pk.h:763]: Style: missing space before `const` (sorry).

File: library/pkwrite.c [hunk 2]
```diff
@@ -310,7 +310,7 @@ int mbedtls_pk_write_pubkey_der( mbedtls_pk_context *key, unsigned char *buf, si
     return( (int) len );
 }
 
-int mbedtls_pk_write_key_der( mbedtls_pk_context *key, unsigned char *buf, size_t size )
+int mbedtls_pk_write_key_der(const mbedtls_pk_context *key, unsigned char *buf, size_t size )
```
Comment (mpg) [inline library/pkwrite.c:313]: Style: same.

**Review by mpg — CHANGES_REQUESTED** [review]
Looks good to me except for two style issues.

**Review by mpg — APPROVED** [review]
Thanks for fixing the style issues. Looks all good to me now.

**Review by chris-jones-arm — APPROVED** [review]
Looks good to me.

**Comment by chris-jones-arm** [comment]
Before we merge, are we waiting for the other const PR's and the changelog PR or are we merging as we go?

**Comment by mpg** [comment]
No I think we can merge const PRs as we go, and then merge the ChangeLog PR last.

**Pull request merged**